### ISSUE TITLE
Support JWK endpoint refresh if kid is missing

### ISSUE THREAD

**Issue by rq30** [istio/istio#41588]
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
JWK keys are updated only after a certain interval (20 mins default) have passed. If the  oauth server issues a token with a new 'kid' it cannot be used until the refresh interval expires and keys are fetched again. This behaviour is undesirable as the server should be able to rotate the keys at any time. Ideally along with the refresh interval if a token is encountered whose kid is not in the cache it should be fetched from JWK endpoint. 
https://openid.net/specs/openid-connect-core-1_0.html#RotateSigKeys.

**Describe alternatives you've considered**
Reducing the refresh interval won't fix the issue. 

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
This is essentially impossible if I understand correct for the case when istiod is fetching, as envoy is the one that knows about mismatched kid  and istiod is fetching it. There is an option to have envoy also do the fetching which would make. this plausible - I'm not sure if it has this behavior though
